src/rollover.py

```diff
@@ -486,24 +477,55 @@ def main_rollover(args):
     return return_value
 
 
-def main_docker_stop(args):
+def run_with_timeout(instance_id, command, timeout):
     """
-    Main entry point for the docker-stop command
+        @param instance_id str
+        @param command str
+        @param timeout int
+        @return bool if command succeeded
     """
-    ec2_client = ec2.EC2Client()
-    info = ec2_client.describe_instances([args.ec2_id])
-    return ssh.stop_all_containers(info[args.ec2_id]['PrivateIpAddress'],
-                                   args.timeout)
 
+    client = boto3.client('ssm')
+    response = client.send_command(
+        InstanceIds = [instance_id],
+        DocumentName = 'AWS-RunShellScript',
+        TimeoutSeconds = max(timeout, 30),
+        Comment = '',
+        Parameters = {
+            'commands': ["#!/bin/bash", command]
+        },
+        OutputS3BucketName = 'clever-test',
+        OutputS3KeyPrefix = 'test-rollover'
+    )
+
+    # Error sending response
+    if response.get('ResponseMetadata', {}).get('HTTPStatusCode') is not 200:
+        print "ERROR sending command:", response
+        return False
+
+    command_id = response.get('Command').get('CommandId')
+
+    # Not sure when/if this actually happens; adding as a safeguard for now.
+    if not command_id:
+        return False
```

Comment:
I'm going to take this out since it should never happen (I only had an error because I had a typo in the JSON key name). I'm fairly certain that any command that is successfully _issued_ gets assigned an ID, even if the command doesn't successfully complete.